### PR TITLE
Improvements: Extract ComponentIdProvider

### PR DESCRIPTION
**Pull request by lciolecki**

### CODE REVIEW

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/component/DefaultComponentIdProviderSpec.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      (componentsType, componentNameToOverride, componentsType.map(_ => overriddenId)),
+    )
+
+    forAll(testingData) { (componentsType: List[ComponentType], name: String, expected: List[ComponentId]) =>
+      val result = componentsType.map(componentIdProvider.createComponentId(processingType, name, _))
+      result shouldBe expected
+    }
+  }
+
+  it should "create ComponentId for NodeData" in {
```
Comment (mproch) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/component/DefaultComponentIdProviderSpec.scala:55]: I wonder, if we moved implementation to this class, and have tests added here, shouldn't we remove/simplify some tests in ComponentService?

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentsUiConfigExtractor.scala [hunk 2]
```diff
@@ -25,7 +25,7 @@ object ComponentsUiConfigExtractor {
     OptionReader
       .optionValueReader[String]
       .read(config, path)
-      .map(ComponentId.create)
+      .map(ComponentId.apply)
```
[the remark this answers is not in the document]
Comment (lciolecki) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentsUiConfigExtractor.scala:28]: I can do `.map(ComponentId(_))` but what the difference?

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentIdProvider.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package pl.touk.nussknacker.ui.component
+
+import pl.touk.nussknacker.engine.api.component.ComponentType.ComponentType
+import pl.touk.nussknacker.engine.api.component.{ComponentId, ComponentType}
+import pl.touk.nussknacker.engine.graph.node.{NodeData, WithComponent}
+import pl.touk.nussknacker.restmodel.process.ProcessingType
+
+trait ComponentIdProvider {
+  def createComponentId(processingType: ProcessingType, name: String, componentType: ComponentType): ComponentId
+  def forBaseComponent(componentType: ComponentType): ComponentId
```
[the remark this answers is not in the document]
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentIdProvider.scala]: Right now it looks different - this comment is outdated.

[the remark this answers is not in the document]
Comment (mproch) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentIdProvider.scala]: true...

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentsUiConfigExtractor.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (mproch) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentsUiConfigExtractor.scala:28]: ok, if the second doesn't work, then...

**Review by mproch — APPROVED** [review]
:+1:

**Pull request merged**